prow/crier/reporters/gerrit/reporter.go

```diff
@@ -341,7 +347,11 @@ func (c *Client) Report(ctx context.Context, logger *logrus.Entry, pj *v1.ProwJo
 				mostRecentJob[pjOnRevisionWithSameLabel.Spec.Job] = &pjsOnRevisionWithSameLabel.Items[idx]
 			}
 		}
-		for _, pjOnRevisionWithSameLabel := range mostRecentJob {
+		for jobName, pjOnRevisionWithSameLabel := range mostRecentJob {
+			if jobName == pj.Spec.Job && pjOnRevisionWithSameLabel.CreationTimestamp.After(pj.CreationTimestamp.Time) {
+				logger.WithField("newer_job", pjOnRevisionWithSameLabel.Name).Debug("There is a newer job, skip reporting.")
+				return nil, nil, nil
+			}
```

Comment:
Is this needed? I would have expected the `mostRecentJob` map to already include `pj` as the most recent job. If `pj` is not the most recent job we wouldn't be reporting.

Reply:
yeah this is the expected behavior and is what this piece of code does. Did you mean to do this check earlier?

Reply:
No, I'm confused by the purpose of this. It looks like dead code to me. We should never have any job with the same name as `pj` in `mostRecentJob` except for `pj` itself, right?

Reply:
I see, you're right, `ShouldReport` already does this so yeah this is dead code, let me delete it